annotated_types/__init__.py

```diff
@@ -106,55 +137,67 @@ class BaseMetadata:
 
 
 @dataclass(frozen=True, **SLOTS)
-class Gt(BaseMetadata):
+class Gt(BaseMetadata, Generic[T]):
     """Gt(gt=x) implies that the value must be greater than x.
 
     It can be used with any type that supports the ``>`` operator,
     including numbers, dates and times, strings, sets, and so on.
     """
 
-    gt: SupportsGt
+    gt: T
+
+    def __supports_type__(self, obj: SupportsGt[T]) -> bool:
+        return obj > self.gt
```

Comment:
I had to flip all of these around. On the one hand this means that `Gt(object)` is now valid, but in theory that's okay:

```python
from typing import Annotated

from annotated_types import Gt


class MyObjectComparer:
    def __gt__(self, obj: object) -> bool:
        return id(obj) > id(self)

assert MyObjectComparer() > object()  # okay

Annotated[MyObjectComparer, Gt(object())]  # okay
Annotated[int, Gt(object())]  # not okay
Gt(object()).__supports_type__(MyObjectComparer())  # okay
Gt(object()).__supports_type__(int)  # not okay
```